build_defs/build_defs.bzl

```diff
@@ -39,17 +44,60 @@ def arcs_cc_schema(name, src, out = None):
         file_extension = ".h",
     )
 
-def arcs_kt_schema(name, src, out = None):
+def _wasm_name(name):
+    return name + "_wasm"
+
+def _wasm_deps(deps):
+    return [_wasm_name(dep) for dep in deps]
+
+def _jvm_and_wasm_library(name, srcs, deps):
+    kt_android_library(
+        name = name,
+        srcs = srcs,
+        deps = deps
+    )
+
+    kt_wasm_library(
+        name = _wasm_name(name),
+        srcs = srcs,
+        deps = _wasm_deps(deps)
+    )
+
+def arcs_kt_schema(name, src, out = None, deps = [], wasm_deps = []):
     """Generates a Kotlin file for the given .arcs schema file."""
+    out = out or _output_name(src, ".kt")
     _run_schema2pkg(
-        name = name,
+        name = name + "_genrule",
         src = src,
         out = out,
         language_name = "Kotlin",
         language_flag = "--kotlin",
         file_extension = ".kt",
     )
 
+    return out
+
+def arcs_manifest(name, srcs, deps = []):
+    """Generates Kotlin+Wasm Library targets from .arcs file translated to Kotlin."""
+    # TODO: when converted to rule()+provider() verify deps are arcs_manifest rules
+    outs = [arcs_kt_schema(_output_name(src), src, deps) for src in srcs]
+    deps = ["//src/wasm/kotlin:arcs"] + deps
+
+    _jvm_and_wasm_library(
```

Comment:
Similarly here, are you sure we always want to be generating both jvm and wasm code? Seems wasteful if you're only writing code for one of those two.

Reply:
It doesn't actually generate anything, unless he targets end up in the final action graph. The only real impact is during blaze/bazel preprocess/analysis phase, since it has to analyze 2x targets.

Reply:
Disabled JVM targets for now